E0110 needs to be updated to new format

From: src/test/compile-fail/E0110.rs

Error E0110 needs a span_label, updating it from:

```
error[E0110]: lifetime parameters are not allowed on this type
  --> src/test/compile-fail/E0110.rs:11:14
   |
11 | type X = u32<'static>; //~ ERROR E0110
   |              ^^^^^^^
```

To:

```
error[E0110]: lifetime parameters are not allowed on this type
  --> src/test/compile-fail/E0110.rs:11:14
   |
11 | type X = u32<'static>; //~ ERROR E0110
   |              ^^^^^^^ lifetime parameter not allowed on this type
```

--- a/src/librustc/middle/astconv_util.rs
+++ b/src/librustc/middle/astconv_util.rs
@@ -29,8 +29,11 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                 break;
             }
             for lifetime in segment.parameters.lifetimes() {
-                span_err!(self.sess, lifetime.span, E0110,
-                          "lifetime parameters are not allowed on this type");
+                struct_span_err!(self.sess, lifetime.span, E0110,
+                                 "lifetime parameters are not allowed on this type")
+                    .span_label(lifetime.span,
+                                &format!("lifetime parameter not allowed on this type"))
+                    .emit();
                 break;
             }
             for binding in segment.parameters.bindings() {

--- a/src/test/compile-fail/E0110.rs
+++ b/src/test/compile-fail/E0110.rs
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 type X = u32<'static>; //~ ERROR E0110
+                       //~| NOTE lifetime parameter not allowed on this type
 
 fn main() {
 }

